extensions/blocks/podcast-player/view.js

```diff
@@ -44,23 +48,143 @@ window.jetpackPodcastPlayers = {
+	const audioEl = e.detail.target;
+	const parentBlockEl = audioEl.closest( '.wp-block-jetpack-podcast-player' );
+	if ( ! parentBlockEl ) {
+		return;
+	}
+
+	// Clean up any error indication if present
+	const episodeErrorEl = parentBlockEl.querySelector( '.podcast-player__episode-error' );
+	if ( episodeErrorEl ) {
+		parentBlockEl.classList.remove( 'is-error' );
+		episodeErrorEl.remove();
+	}
+
+	parentBlockEl.classList.remove( 'is-paused' );
+	parentBlockEl.classList.add( 'is-playing' );
+}
+
+function handleMediaPause( e ) {
+	const audioEl = e.detail.target;
+	const parentBlockEl = audioEl.closest( '.wp-block-jetpack-podcast-player' );
+
+	parentBlockEl.classList.remove( 'is-playing' );
+	parentBlockEl.classList.add( 'is-paused' );
+}
+
+function handleMediaError( e ) {
+	const audioEl = e.detail.target;
+	const parentBlockEl = audioEl.closest( '.wp-block-jetpack-podcast-player' );
+	const activeEpisodeLinkEl = parentBlockEl.querySelector( '.is-active > a' );
+
+	renderEpisodeError( activeEpisodeLinkEl );
+}
+
+const episodeLinkEls = document.querySelectorAll( '[data-jetpack-podcast-audio]' );
+
+Array.prototype.forEach.call( episodeLinkEls, buildEpisodeLinkClickHandler );
+
+function buildEpisodeLinkClickHandler( episodeLinkEl ) {
+	episodeLinkEl.addEventListener( 'click', handleEpisodeLinkClick );
+	episodeLinkEl.addEventListener( 'keydown', handleEpisodeLinkKeydown );
+}
+
+function handleEpisodeLinkKeydown( e ) {
+	// we only need to track spacebar, as Enter is already handled by the browser since it's an `<a>` element
+	if ( event.key === ' ' ) {
+		handleEpisodeLinkClick( e );
+	}
+}
+
+function handleEpisodeLinkClick( e ) {
 	// Prevent handling clicks if a modifier is in use.
 	if ( e.shiftKey || e.metaKey || e.altKey ) {
 		return;
 	}
 
-	// Check if the clicked element was episode link.
-	const audioUrl = e.target.getAttribute( 'data-jetpack-podcast-audio' );
-	if ( audioUrl ) {
-		const block = e.target.closest( '.wp-block-jetpack-podcast-player' );
-		const player = block && playerInstances[ block.id ];
-		if ( player ) {
-			player.audio.pause();
-			player.audio.src = audioUrl;
-			player.audio.play();
-			e.preventDefault();
-		}
+	e.preventDefault();
+
+	const episodeLinkEl = e.currentTarget;
+
+	// Get clicked episode audio URL
+	const audioUrl = episodeLinkEl.getAttribute( 'data-jetpack-podcast-audio' );
+	if ( ! audioUrl ) {
+		return renderEpisodeError( episodeLinkEl );
+	}
+
+	// Get clicked episode element
+	const episodeEl = episodeLinkEl.closest( '.podcast-player__episode' );
+	if ( ! episodeLinkEls ) {
+		return renderEpisodeError( episodeLinkEl );
+	}
+
+	// Get episode's parent block element
+	const blockEl = episodeEl.closest( '.wp-block-jetpack-podcast-player' );
+	if ( ! blockEl ) {
+		return renderEpisodeError( episodeLinkEl );
+	}
+
+	// Get player instance by block id
+	const player = playerInstances[ blockEl.id ];
+	if ( ! player ) {
+		return renderEpisodeError( episodeLinkEl );
+	}
+
```

Comment:
This may be on your roadmap already, but before pausing and switching the episode, we should check to see if the episode is already playing. If it's already playing, I'd think we should just *pause the current episode*, not restart it. Likewise if the episode is the current item but paused, we should just *play* it rather than restarting.

Reply:
Great catch, I was wondering how to handle it! 🙌

Reply:
Should this be in a follow-up PR?

Reply:
Yes, please!

Reply:
Sorry, it's been done already 🙈